src/ComposerRequireChecker/Cli/ResultsWriter/BatchWriter.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ComposerRequireChecker\Cli\ResultsWriter;
+
+use function array_filter;
+
+final class BatchWriter implements ResultsWriter
```

Comment:
More than `BatchWriter`, this is `WriteResultsToMultipleWriters` (batching is a separate concept)

Reply:
This class requires some tests (mostly mock based)